File: 240504/최소 와이파이 수/minimum-number-of-wifi.py
N, M = map(int, input().split())
people = list(map(int, input().split()))

def new_idx(x, n_idx):
    for i in range(N):
        if i >= n_idx and people[i] == 1:
            return i
        elif i == N - 1:
            return n_idx
        else:
            continue

def is_range(x):
    if x >= N:
        return False
    elif people[x] == 0:
        return True
    else:
        return False

cnts = 0
if 1 not in people:
    print(cnts)
else:
    idx = 0
    while idx < N:
        if people[idx: idx + 2 * M + 1].count(1) != 0:
            cnts += 1

        if people[idx: idx + 2 * M + 1].count(1) == 0 or is_range(idx + 2 * M + 1):
            now_idx = idx + 2 * M + 1
            idx = new_idx(people, now_idx)
        else:
            idx += 2 * M + 1

print(cnts)

File: 260430/최소 와이파이 수/minimum-number-of-wifi.cpp
#include <bits/stdc++.h>
using namespace std;

int N, M;
vector<int> people;

int new_idx(int n_idx) 
{
    for (int i = 0; i < N; i++) 
    {
        if (i >= n_idx && people[i] == 1) 
        {
            return i;
        } 
        else if (i == N - 1) 
        {
            return n_idx;
        } 
        else 
        {
            continue;
        }
    }
    return n_idx;
}

bool is_range(int x) 
{
    if (x >= N) 
    {
        return false;
    } 
    else if (people[x] == 0) 
    {
        return true;
    } 
    else 
    {
        return false;
    }
}

int count_one(int l, int r) 
{
    int cnt = 0;
    r = min(r, N);

    for (int i = l; i < r; i++) 
    {
        if (people[i] == 1) 
        {
            cnt++;
        }
    }

    return cnt;
}

int main() 
{
    // Please write your code here.
    cin >> N >> M;

    people.resize(N);
    for (int i = 0; i < N; i++) 
    {
        cin >> people[i];
    }

    int cnts = 0;

    bool has_one = false;
    for (int x : people) 
    {
        if (x == 1) 
        {
            has_one = true;
            break;
        }
    }

    if (!has_one) 
    {
        cout << cnts << '\n';
    } 
    else 
    {
        int idx = 0;

        while (idx < N) 
        {
            int ones = count_one(idx, idx + 2 * M + 1);

            if (ones != 0) 
            {
                cnts++;
            }

            if (ones == 0 || is_range(idx + 2 * M + 1)) 
            {
                int now_idx = idx + 2 * M + 1;
                idx = new_idx(now_idx);
            } 
            else 
            {
                idx += 2 * M + 1;
            }
        }
        cout << cnts << '\n';
    }
    return 0;
}
